crates/ironrdp-client/src/rdp.rs

```diff
@@ -107,7 +114,9 @@ async fn connect(
 
     let mut connector = connector::ClientConnector::new(config.connector.clone())
         .with_server_addr(server_addr)
-        .with_static_channel(ironrdp::dvc::DrdynvcClient::new())
+        .with_static_channel(
+            ironrdp::dvc::DrdynvcClient::new().with_dynamic_channel(DisplayControlClient::new(|_| Ok(Vec::new()))),
```

Comment:
From our doc comment:
```
    /// The `callback` will be called when capabilities are received from the server.
    /// It is important to note that the channel will not be fully operational until the capabilities are received.
    /// Attempting to send messages before the capabilities are received will result in an error or a silent failure.
```

I think we need to block any dynamic resolution changes from the client until this callback (passed in new) receives some caps. Also, based on caps we should prevent setting resolution to values greater than max monitor area received in caps.

Reply:
> I think we need to block any dynamic resolution changes from the client until this callback (passed in new) receives some caps.

This is possibility is already handled here:

https://github.com/Devolutions/IronRDP/blob/4d878fbcf6608dcb19161ea62e2f36cffd8a28eb/crates/ironrdp-client/src/rdp.rs#L207-L213

I added this callback for Teleport purposes, where we do things a little differently:
1. Any resizes that occur before the channel is open are held in an internal register
2. The callback we pass here reads that register and returns the encoded resize, such that the most recent browser size gets sent as soon as the channel is ready for it.

Something similar to that could be implemented in IronRDP, but it's a substantial piece of work compared to what's there now (reconnect if the channel isn't ready).

> Also, based on caps we should prevent setting resolution to values greater than max monitor area received in caps.

Seems like a fine check but in practice it's not a big risk, I just added a TODO for this for now 1644ff717ae8294f9f155b4e510bbf9024aa2467